meetings/2024/LDM-2024-04-15.md

```diff
@@ -0,0 +1,66 @@
+# C# Language Design Meeting for April 15, 2024
+
+## Agenda
+
+- [Non-enumerable collection types](Julien)
+  - (https://github.com/dotnet/csharplang/issues/7744) - championed issue
+  - (https://github.com/dotnet/csharplang/pull/7895) - requested details
+- Interceptors [location specifier](https://github.com/dotnet/csharplang/blob/interceptors-2024-01/proposals/interceptors-issues-2024-01.md#add-interceptslocationstring-locationspecifier) (Rikki)
+- [Relax `Add` requirementfo collection expression conversion](https://github.com/dotnet/csharplang/issues/8034)
+
+## Non-enumerable collection types
+
+Should collections that  are not enumerable be usable in parameter collections, if they have a CollectionBuilder attribute that defines a single Create method?
+
+In this case, the element type of the collection can be determined from the Create method if it is not already known.
+
+Changes to the spec had been previously discussed and the details changes were reviewed, as shown in #7895 and #7744. This work has been merged due to time constraints and this discussion was to confirm approval.
+
+### Conclusion
+
+These proposed changes were approved.
+
+## Interceptors
+
+### File location
+
+Interceptors must identify the location in code where interception occurs. The issue is how to identify that location for use in the interceptor attribute.
+
+Things that have been tried or considered for identifying the file:
+
+* Absolute paths, per the preview, are bad for portability.
+* Relative paths required the context be passed around or a project base path.
+
+The proposed arguments to `[InterceptsLocation]` are a version number and an opaque data string. The opaque data string is not intended to be human readable. It is expected that this string will be inserted by a generator that retrieves it via a call into the interceptors API that requests an interceptible location.
+
+The proposal is to create the opaque string by combining a checksum of the file contents, an integer location of the call in the syntax and a string file name for error reporting. Hand authoring interceptor locations is an unsupported scenario.
+
+A version identifier is included to allow future change to be made in the calculation of the data string.
+
+### Other public API changes
+
+The team also mentioned an that there is an experimental public API that lets you determine whether syntax is being intercepted and whose intercepting. This might be used in an analyzer, for example.
```

Comment:
```suggestion
The team also mentioned that there is an experimental public API that lets you determine whether syntax is being intercepted and who is intercepting. This might be used in an analyzer, for example.
```